.gitlab-ci.yml

```diff
@@ -170,6 +173,51 @@ fail_on_non_triggered_tag:
     - echo CI_PIPELINE_SOURCE=$CI_PIPELINE_SOURCE CI_COMMIT_TAG=$CI_COMMIT_TAG
     - '[ "$CI_COMMIT_TAG" == "" -o "$CI_PIPELINE_SOURCE" != "push" ]'
 
+#
+# deps_build
+#
+#
+
+empty:
+  stage: deps_build
+  tags: [ "runner:main", "size:large" ]
+  before_script: []
+  script:
+    - echo "This is an empty step that can be used as a dependency for jobs that depend on nothing"
+    - echo "This can be removed once https://gitlab.com/gitlab-org/gitlab/issues/30631 is implemented"
+
+# build libbcc
+.build_libbcc_common:
+  stage: deps_build
+  before_script: []
+  script:
+    - git clone -b "$BCC_VERSION" --depth=1 https://github.com/iovisor/bcc.git /tmp/bcc
+    - mkdir /tmp/bcc/build
+    - cd /tmp/bcc/build
+    - cmake .. -DCMAKE_INSTALL_PREFIX=/opt/libbcc -DCMAKE_EXE_LINKER_FLAGS='-Wl,-rpath,/opt/datadog-agent/embedded/lib' -DCMAKE_SHARED_LINKER_FLAGS='-Wl,-rpath,/opt/datadog-agent/embedded/lib'
```

Comment:
Could this be built into a new build image? Doing this on every build introduces a build time dependency on github and is slow (compiling bcc anecdotally takes a minute on my laptop)

Reply:
All the dependencies of the agent are currently built by omnibus in the agent pipeline.
Here is an example of an agent pipeline logs: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/25136857
We can see behind the:
```
[Builder: procps-ng] …
[Builder: zlib] …
[Builder: cacerts] …
[Builder: xproto] …
[Builder: util-macros] …
[Builder: libiconv] …
… many more …
```
the logs of the build of many 3rd party dependencies.

The specificity of `libbcc` is that it requires a different build environment. So, it is built in its own job.
But if we want to build dependencies in a distinct pipeline in order to speed up the agent one, shouldn’t we do that for all the 3rd party dependencies and not only for `libbcc` ?

Reply:
Because of the way the dependencies are described, the new `build_libbcc_{arm64,x64}` jobs are run in parallel to the `run_tests_*` ones.
Both sets of jobs are required to move to the `Binary_build` stage.
And `build_libbcc_{arm64,x64}` jobs are faster than the `run_tests_*` ones.
So, in the end, adding those extra jobs doesn’t change the total time it takes for the whole pipeline to complete because it runs in parallel to existing jobs and they are not on the critical path.